conflicting implementations of trait `std::convert::From<gpu::vk::NullHandle>` for type `u64`

When building with the Vulkan feature and targeting Android I see a conflicting implementations error. In this case, VKDeviceMemory, VKBuffer, and VKImage are all type alias for u64, so the compiler gets mad for conflicting implementations.

```
error[E0119]: conflicting implementations of trait `std::convert::From<gpu::vk::NullHandle>` for type `u64`:
  --> /Users/marco/.cargo/registry/src/github.com-1ecc6299db9ec823/skia-safe-0.21.0/src/gpu/vk.rs:67:1
   |
49 | impl From<NullHandle> for Buffer {
   | -------------------------------- first implementation here
...
67 | impl From<NullHandle> for DeviceMemory {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `u64`

error[E0119]: conflicting implementations of trait `std::convert::From<gpu::vk::NullHandle>` for type `u64`:
  --> /Users/marco/.cargo/registry/src/github.com-1ecc6299db9ec823/skia-safe-0.21.0/src/gpu/vk.rs:85:1
   |
49 | impl From<NullHandle> for Buffer {
   | -------------------------------- first implementation here
...
85 | impl From<NullHandle> for Image {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `u64`

error[E0119]: conflicting implementations of trait `std::convert::From<gpu::vk::NullHandle>` for type `u64`:
  --> /Users/marco/.cargo/registry/src/github.com-1ecc6299db9ec823/skia-safe-0.21.0/src/gpu/vk.rs:97:1
   |
```

--- a/azure-pipelines-template.yml
+++ b/azure-pipelines-template.yml
@@ -153,6 +153,11 @@ jobs:
 
     - template: 'azure-pipelines-build-target.yml'
       parameters:
+        target: 'i686-linux-android'
+        androidEnv: true
+
+    - template: 'azure-pipelines-build-target.yml'
+      parameters:
         target: 'aarch64-apple-ios'
         releaseBinaries: ${{ parameters.deployRelease }}
 
@@ -170,6 +175,11 @@ jobs:
     - template: 'azure-pipelines-build-target.yml'
       parameters:
         target: 'x86_64-linux-android'
+        androidEnv: true
+
+    - template: 'azure-pipelines-build-target.yml'
+      parameters:
+        target: 'i686-linux-android'
         androidEnv: true
 
 # TODO: Windows Android builds are temporarily disabled, because of:

--- a/skia-bindings/build_support/android.rs
+++ b/skia-bindings/build_support/android.rs
@@ -18,10 +18,9 @@ pub fn additional_clang_args(target: &str, target_arch: &str) -> Vec<String> {
     };
 
     match (cargo::host().system.as_str(), target_arch) {
-        // under Linux hosts and x68 targets, we do have better chances without
+        // On Linux hosts (Ubuntu 18, LLVM 6) and x86_64 targets, we do have better chances without
         // any further configuration adjustments.
-        // ("linux", "i686") => {}
-        // ("linux", "x86_64") => {}
+        ("linux", "x86_64") => {}
         _ => {
             let ndk = ndk();
 

--- a/skia-bindings/build_support/android.rs
+++ b/skia-bindings/build_support/android.rs
@@ -20,8 +20,8 @@ pub fn additional_clang_args(target: &str, target_arch: &str) -> Vec<String> {
     match (cargo::host().system.as_str(), target_arch) {
         // under Linux hosts and x68 targets, we do have better chances without
         // any further configuration adjustments.
-        ("linux", "i686") => {}
-        ("linux", "x86_64") => {}
+        // ("linux", "i686") => {}
+        // ("linux", "x86_64") => {}
         _ => {
             let ndk = ndk();
 

--- a/skia-bindings/build_support/skia.rs
+++ b/skia-bindings/build_support/skia.rs
@@ -485,18 +485,22 @@ impl BinariesConfiguration {
         }
     }
 
-    /// Inform cargo that the output files of the given configuration are available and
+    /// Inform cargo that the library files of the given configuration are available and
     /// can be used as dependencies.
     pub fn commit_to_cargo(&self) {
-        cargo::add_link_libs(&self.link_libraries);
         println!(
             "cargo:rustc-link-search={}",
             self.output_directory.to_str().unwrap()
         );
 
+        // On Linux, the order is significant, first the static libraries we built, and then
+        // the system libraries.
+
         for lib in &self.built_libraries {
             cargo::add_link_lib(format!("static={}", lib));
         }
+
+        cargo::add_link_libs(&self.link_libraries);
     }
 }
 

--- a/skia-safe/src/gpu/vk.rs
+++ b/skia-safe/src/gpu/vk.rs
@@ -46,6 +46,7 @@ pub const QUEUE_FAMILY_IGNORED: u32 = !0;
 pub struct NullHandle;
 pub const NULL_HANDLE: NullHandle = NullHandle;
 
+#[cfg(target_pointer_width = "64")]
 impl From<NullHandle> for Buffer {
     fn from(_: NullHandle) -> Self {
         ptr::null_mut()
@@ -64,6 +65,7 @@ impl From<NullHandle> for Device {
     }
 }
 
+#[cfg(target_pointer_width = "64")]
 impl From<NullHandle> for DeviceMemory {
     fn from(_: NullHandle) -> Self {
         ptr::null_mut()
@@ -82,6 +84,7 @@ impl From<NullHandle> for PhysicalDevice {
     }
 }
 
+#[cfg(target_pointer_width = "64")]
 impl From<NullHandle> for Image {
     fn from(_: NullHandle) -> Self {
         ptr::null_mut()
@@ -94,8 +97,16 @@ impl From<NullHandle> for Queue {
     }
 }
 
+#[cfg(target_pointer_width = "64")]
 impl From<NullHandle> for RenderPass {
     fn from(_: NullHandle) -> Self {
         ptr::null_mut()
+    }
+}
+
+#[cfg(not(target_pointer_width = "64"))]
+impl From<NullHandle> for u64 {
+    fn from(_: NullHandle) -> Self {
+        0
     }
 }

